pkg/jobs/registry.go

```diff
@@ -680,6 +680,26 @@ func (r *Registry) LoadJob(ctx context.Context, jobID jobspb.JobID) (*Job, error
 	return r.LoadJobWithTxn(ctx, jobID, nil)
 }
 
+// LoadJob loads an existing job with the given jobID from the
+// system.jobs table. The job must be claimed by the given sessionID
+// or an error is returned.
+func (r *Registry) LoadJobWithSessionID(
+	ctx context.Context, jobID jobspb.JobID, sessionID sqlliveness.SessionID,
+) (*Job, error) {
```

Comment:
I understand why we're doing this (i.e. passing in the sessionID to this method).
I am concerned, however, is that the API is getting even more subtle (and brittle): which method do you use (LoadJob or LoadJobWithTxn or now LoadJobWithSessionID).

Furthermore, because we had to change protocol message specifications, it appears that this change would actually be less backportable than a change that didn't change any protocol messages.

My prefer mechanism would probably *not* be backportable since it would likely break some users; and that mechanism would have been to have `LoadJob` method consult `r.adoptedJobs` map to pick up correct session ID, and to set session ID to an invalid value if job is not in the local registry (i.e. force job updates to be done on the coordnator.)

And alternative, which I feel would be backportable would be to have something like:
`LoadAdoptedJob` (or perhaps `LoadLocallyExecutingJob`).  Same deal, but return an error if job is not adopted by local registry, or set session id to an invalid value so that updates fail.

I think with such a change, we won't have to change any proto messages.

Thoughts?

Reply:
> Furthermore, because we had to change protocol message specifications, it appears that this change would actually be less backportable than a change that didn't change any protocol messages.

I believe that the protobuf changes should be strictly compatible; but we will need to review that carefully.

> And alternative, which I feel would be backportable would be to have something like:
LoadAdoptedJob (or perhaps LoadLocallyExecutingJob). Same deal, but return an error if job is not adopted by local registry, or set session id to an invalid value so that updates fail.

I like this, I'll look into this as I think it should still be rather small and not require threading more data through the proto.